### PR TITLE
Bump dependencies and CI

### PR DESCRIPTION
**Pull request by andreasabel**
- include latest GHCs in CI
- ~~latest minor version is determined by haskell-actions/setup~~
- ~~freeze file is a bad cache key as it contains a time stamp, use the build plan instead (the correct choice)~~

~~Successful CI run at: https://github.com/andreasabel/Haxl/actions/runs/6725224644~~
CI running here: https://github.com/andreasabel/Haxl/pull/1

### CODE REVIEW

**Comment by andreasabel** [comment]
@simonmar : PR ready

**Comment by josefs** [comment]
Hi @andreasabel ! Thanks for the patch! Unfortunately it looks like it's broken for 9.4.

**Comment by andreasabel** [comment]
Updated this PR.  Not much remains, since `test.yml` was deleted on `main`.
Now this just updates the Haskell CI to the latest GHCs.

File: .github/workflows/haskell-ci.yml [hunk 1]
```diff
@@ -6,11 +6,11 @@
 #
 #   haskell-ci regenerate
 #
-# For more information, see https://github.com/haskell-CI/haskell-ci
+# For more information, see https://github.com/andreasabel/haskell-ci
```
Comment (josefs) [inline .github/workflows/haskell-ci.yml:9]: Are you expecting your changes to be upstreamed or are you going to maintain this fork?

**Review by josefs — COMMENTED** [review]
Apologies for dropping this. Thanks for updating this PR.

**Comment by andreasabel** [comment]
GHC 9.10 comes with filepath-1.5, so I relaxed the upper bound.

**Comment by andreasabel** [comment]
CI passes now.

**Pull request closed without merging**